modules/patterns/pages/running-user-scripts-on-the-build-pipeline.adoc

```diff
@@ -0,0 +1,107 @@
++        - name: SCRIPT_RUNNER_IMAGE <1>
++          value: quay.io/my-script-runner-image:latest@sha256:digest
++        - name: SCRIPT
++          value: my-script <2>
++        - name: HERMETIC
++          value: $(params.hermetic)
++        - name: SOURCE_ARTIFACT
++          value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
++      runAfter:
++      - prefetch-dependencies
++      taskRef:
++        params:
++        - name: name
++          value: run-script-oci-ta
++        - name: bundle
++          value: quay.io/konflux-ci/tekton-catalog/task-run-script-oci-ta:0.1@sha256:c0f627069353ebd6d1ed03c8657e281eaf11be63706ea38cc53caf16cf4ffd65
++        - name: kind
++          value: task
++        resolver: bundles
+----
+
+<1> Use `SCRIPT_RUNNER_IMAGE` parameter to specify the container image you want to use to run the script. The image must already ship all the dependencies and additional tools required to run the script.
+
+<2> Use `SCRIPT` parameter to specify the script to run. This can be a command using an absolute path on the container image (`make build`) , a command using the relative path of an script on the source repository (`./my-script.sh build`) or an inline `sh` script. Refer to link:https://github.com/konflux-ci/build-definitions/blob/main/task/run-script-oci-ta/0.1/README.md[the task documentation] for more examples.
+
+=== Modify build-images task
+
+You need to mode `build-images` task to consume the output trusted artifact of the `run-script` task and inject the `SCRIPT_RUNNER_IMAGE` into the sbom file, and run the build images task after the `run-script` one:
+
+[source, diff]
+----
+       - name: SOURCE_ARTIFACT
+-        value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
++        value: $(tasks.run-script.results.SCRIPT_ARTIFACT)
+       - name: CACHI2_ARTIFACT
+         value: $(tasks.prefetch-dependencies.results.CACHI2_ARTIFACT)
+       - name: IMAGE_APPEND_PLATFORM
+         value: "true"
++      - name: ADDITIONAL_BASE_IMAGES
++        value:
++          - $(tasks.run-script.results.SCRIPT_RUNNER_IMAGE_REFERENCE)
+       runAfter:
+-      - prefetch-dependencies
++      - run-script
+----
+
+=== Modify push-dockerfile task if necessary
+
+When the script is the tool generating the `Containerfile`, you need to modify `push-dockerfile` task to consume the output trusted artifact from `run-script`:
+
+[source, diff]
+----
+       - name: CONTEXT
+         value: $(params.path-context)
+       - name: SOURCE_ARTIFACT
+-        value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
++        value: $(tasks.run-script.results.SCRIPT_ARTIFACT)
+       runAfter:
+       - build-image-index
+       taskRef:
+----
+
+=== Specific use case examples
+
+==== ansible-builder to build Ansible Execution Environments
+
+A link:https://github.com/Zokormazo/konflux-execution-environment-example[sample repository] has been prepared to show how to use this task to build Ansible Execution Environments.
```

Comment:
The refinement doc asks for complete examples. I have this one here, but is weird to link from docs to my personal github organization :)

I have plans to create another example with backstage dynamic plugins OCI images, but won't have time for that this week

Reply:
I changed the link to the PR for now. I like the idea of a monorepo of examples on konflux-ci/ . We can move on that direction once we have multiple examples :)